optaplanner-core/src/test/java/org/optaplanner/core/impl/heuristic/selector/move/generic/PillarChangeMoveTest.java

```diff
@@ -116,6 +119,52 @@ public void doMove() {
         assertEquals(v2, c.getValue());
     }
 
+    @Test
+    public void rebase() {
+        GenuineVariableDescriptor<TestdataSolution> variableDescriptor = TestdataEntity.buildVariableDescriptorForValue();
+
+        TestdataValue v1 = new TestdataValue("v1");
+        TestdataValue v2 = new TestdataValue("v2");
+        TestdataValue v3 = new TestdataValue("v3");
+        TestdataEntity e1 = new TestdataEntity("e1", v1);
+        TestdataEntity e2 = new TestdataEntity("e2", null);
+        TestdataEntity e3 = new TestdataEntity("e3", v1);
+        TestdataEntity e4 = new TestdataEntity("e4", v3);
+
+        TestdataValue destinationV1 = new TestdataValue("v1");
+        TestdataValue destinationV2 = new TestdataValue("v2");
+        TestdataValue destinationV3 = new TestdataValue("v3");
+        TestdataEntity destinationE1 = new TestdataEntity("e1", destinationV1);
+        TestdataEntity destinationE2 = new TestdataEntity("e2", null);
+        TestdataEntity destinationE3 = new TestdataEntity("e3", destinationV1);
+        TestdataEntity destinationE4 = new TestdataEntity("e4", destinationV3);
+
+        ScoreDirector<TestdataSolution> destinationScoreDirector = mockRebasingScoreDirector(
+                variableDescriptor.getEntityDescriptor().getSolutionDescriptor(), new Object[][]{
+                        {v1, destinationV1},
+                        {v2, destinationV2},
+                        {v3, destinationV3},
+                        {e1, destinationE1},
+                        {e2, destinationE2},
+                        {e3, destinationE3},
+                        {e4, destinationE4},
+                });
+
+        assertSameProperties(Arrays.asList(destinationE1, destinationE3), null,
+                new PillarChangeMove<>(Arrays.asList(e1, e3), variableDescriptor, null).rebase(destinationScoreDirector));
+        assertSameProperties(Arrays.asList(destinationE1, destinationE3), destinationV3,
+                new PillarChangeMove<>(Arrays.asList(e1, e3), variableDescriptor, v3).rebase(destinationScoreDirector));
+        assertSameProperties(Arrays.asList(destinationE2), destinationV1,
+                new PillarChangeMove<>(Arrays.asList(e2), variableDescriptor, v1).rebase(destinationScoreDirector));
+        assertSameProperties(Arrays.asList(destinationE1), destinationV2,
+                new PillarChangeMove<>(Arrays.asList(e1), variableDescriptor, v2).rebase(destinationScoreDirector));
+    }
+
+    public void assertSameProperties(List<Object> pillar, Object toPlanningVariable, PillarChangeMove<?> move) {
+        assertListElementsSameExactly(pillar, (List<Object>) move.getPillar());
```

Comment:
Can be replaced by AssertJ's `assertThat(move.getPillar()).hasSameElementsAs(pillar);` but we can change this in bulk later.